### PR TITLE
Move enterprise features to community

### PR DESCRIPTION
**Pull request by erimatnor**
As of this change, a number of enterprise features are accessible to
community users. These features include:

* reorder
* policies around reorder and drop chunks

The move chunks feature is still covered by enterprise. Gapfill no
longer warns about expired enterprise license.

Tests have been updated to reflect these changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,17 +30,17 @@ The major new feature in this release allows users to keep the aggregated
 data in a continuous aggregate while dropping the raw data with drop_chunks.
 This allows users to save storage by keeping only the aggregates.
 
-The semantics of the refresh_lag parameter for continuous aggregates has 
+The semantics of the refresh_lag parameter for continuous aggregates has
```
Comment (mkindahl) [inline CHANGELOG.md]:
```suggestion
The semantics of the `refresh_lag` parameter for continuous aggregates has
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -26,17 +30,17 @@ The major new feature in this release allows users to keep the aggregated
...
 been changed to be relative to the current timestamp instead of the maximum
 value in the table. This change requires that an integer_now func be set on
 hypertables with integer-based time columns to use continuous aggregates on
 this table.
 
-We added a timescaledb.ignore_invalidation_older_than parameter for continuous 
-aggregatess. This parameter accept a time-interval (e.g. 1 month). if set, 
+We added a timescaledb.ignore_invalidation_older_than parameter for continuous
```
Comment (mkindahl) [inline CHANGELOG.md]:
```suggestion
We added a `timescaledb.ignore_invalidation_older_than` parameter for continuous
```

File: tsl/src/bgw_policy/drop_chunks_api.c [hunk 3]
```diff
@@ -106,7 +105,7 @@ drop_chunks_add_policy(PG_FUNCTION_ARGS)
 		{
 			ts_cache_release(hcache);
 			elog(WARNING,
-				 "could not add drop_chunks policy due to existing policy on hypertable with "
+				 "could not add drop chunks policy due to existing policy on hypertable with "
```
Comment (mkindahl) [inline tsl/src/bgw_policy/drop_chunks_api.c]:
```suggestion
				 "could not add drop-chunks policy due to existing policy on hypertable with "
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,10 @@ accidentally triggering the load of a previous DB version.**
 * #1648 Drop chunks for materialized hypertable
 * #1665 Add ignore_invalidation_older_than to timescaledb_information.continuous_aggregates view
 
+**Licensing changes**
+* Gapfill, reorder, and policies around reorder and drop chunks are
```
Comment (svenklemm) [inline CHANGELOG.md]: gapfill is already community

Comment (erimatnor) [reply CHANGELOG.md]: Not according to the code (see diff). 

I guess we could skip mentioning gapfill here if it was already announces as community previously.

File: tsl/test/sql/tsl_tables.sql [hunk 5]
```diff
@@ -14,7 +14,7 @@ RETURNS VOID
 AS :TSL_MODULE_PATHNAME, 'ts_test_bgw_job_delete_by_id'
 LANGUAGE C VOLATILE STRICT;
 
-\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
+SET ROLE :ROLE_DEFAULT_PERM_USER;
```
Comment (gayyappan) [inline tsl/test/sql/tsl_tables.sql]: It is difficult to parse the .sql file (often needed while debugging a regression test) and figure out which roles are getting reset. Keeping this as a specific user makes the role-user explicit. Could you undo these changes and retain the explicit \c <dbname> <user> syntax?

File: tsl/test/sql/tsl_tables.sql [hunk 6]
```diff
@@ -102,13 +102,14 @@ select remove_drop_chunks_policy('test_table');
 -- hypertables that have integer time dimension
 
 select * from _timescaledb_catalog.dimension;
-\c :TEST_DBNAME :ROLE_SUPERUSER
+RESET ROLE;
```
Comment (gayyappan) [inline tsl/test/sql/tsl_tables.sql]: the RESET referred to by the previous comment.

File: tsl/test/sql/tsl_tables.sql (hunk 5 above)
Comment (erimatnor) [reply tsl/test/sql/tsl_tables.sql]: I see what you are saying, and I can undo for this PR.

Still, I do think creating new sessions throughout tests should be avoided for performance reasons but also debugging. It's not possible to run tests through a debugger if you create new sessions all the time. I guess one alternative is to combine the `RESET ROLE` with `SELECT current_user;`?

File: CHANGELOG.md (hunk 4 above)
Comment (svenklemm) [reply CHANGELOG.md]: Gapfill requires community license and your code change only removes a warning and not an actual license check

File: tsl/test/sql/tsl_tables.sql (hunk 5 above)
Comment (gayyappan) [reply tsl/test/sql/tsl_tables.sql]: There are 2 cases where we use different users as part of the test  (1) Some of the tests require testing with different users - primarily for grants and permission testing. ( 2) With the current testing framework, we also do all the setup work (that requires super user permissions) as part of the tests. Resetting roles + show current_user should work in this scenario.

File: CHANGELOG.md (hunk 4 above)
Comment (erimatnor) [reply CHANGELOG.md]: A warning includes a _check_ for a valid enterprise license , and the full message is: `"WARNING:  Timescale License expired"` - `"Your license expired on %s. Renew your license to continue using enterprise features."` Gapfill tests clearly show this warning.

The reason we often aren't enforcing is because we don't want to break the DB when the license expires and instead give the user a chance to renew it. This is true also for the move chunks functionality, which is an enterprise feature that only raises a warning without enforcement (not my addition).

With this in mind, maybe you can point out what you'd like to see changed because I am not actually sure what you are trying to say.

I think you are saying that gapfill is community licensed but the code was never updated to reflect that? If that's the case, I think we should still make a note of this in the changelog because I guess that some gapfill users could have been mislead to believe that they couldn't use gapfill because of the warnings.

File: CHANGELOG.md (hunk 1 above)
Comment (erimatnor) [reply CHANGELOG.md]: I am going to opt out of making changelog content changes since this is not in scope of the PR.

File: tsl/test/sql/tsl_tables.sql (hunk 5 above)
Comment (erimatnor) [reply tsl/test/sql/tsl_tables.sql]: Actually, let me revert this and will adopt a consistent approach in separate PR.

File: CHANGELOG.md (hunk 4 above)
Comment (svenklemm) [reply CHANGELOG.md]: The warning is a generic license expiration warning and not specific to enterprise.

File: CHANGELOG.md [hunk 7]
```diff
@@ -16,6 +16,12 @@ accidentally triggering the load of a previous DB version.**
 * #1668 Cannot add dimension if hypertable has empty chunks
 * #1674 Fix time_bucket_gapfill's interaction with GROUP BY
 
+**Licensing changes**
+* Reorder and policies around reorder and drop chunks are now
+  accessible to community users
+* Gapfill functionality no longer warns about expired enterprise
```
Comment (svenklemm) [inline CHANGELOG.md]: This is not correct because the license expiration warning does not check the license type

Comment (erimatnor) [reply CHANGELOG.md]: Yes, because no other license can expire or should generate this message.

File: CHANGELOG.md [hunk 8]
```diff
@@ -18,6 +18,11 @@ accidentally triggering the load of a previous DB version.**
 * #1686 Fix order by queries on compressed hypertables that have char segment by column
 * #1687 Fix issue with disabling compression when foreign keys are present
 
+**Licensing changes**
+* Reorder and policies around reorder and drop chunks are now
+  accessible to community users
```
Comment (mfreed) [inline CHANGELOG.md]: clarify:  "accessible to community users, not just enterprise"

**Comment by jflambert** [comment]
Is it possible that the docs still need to be updated? I don't see "Community Edition" badges next to most of the [automation policies](https://docs.timescale.com/latest/api#automation-policies) functions.
